pkg/http/listener.go

```diff
@@ -88,6 +89,23 @@ type httpListener struct {
 	errorLogFunc           func(error, string, ...interface{}) // function to be called on errors.
 }
 
+// shouldLogError returns false if error is due to a network timeout or
+// io.EOF and true otherwise
+func shouldLogError(err error) bool {
```

Comment:
Because no logger is present in the package scope (pkg/http), perhaps we could name this function `isUnexpected()`?